lib/Parse/ParsePattern.cpp

```diff
@@ -830,15 +843,25 @@ Parser::parseFunctionSignature(Identifier SimpleName,
     retType = nullptr;
   }
 
-  // Check for 'throws' and 'rethrows' after the type and correct it.
+  // Check for 'throws', 'rethrows' and 'throw' after the type and correct it.
+  // "-> Int throws {" becomes "throws -> Int {"
   if (auto diagOpt = diagnoseInvalidThrows()) {
     assert(arrowLoc.isValid());
     assert(retType);
-    SourceLoc typeEndLoc = Lexer::getLocForEndOfToken(SourceMgr,
-                                                      retType->getEndLoc());
+    SourceLoc typeEndLoc =
+        Lexer::getLocForEndOfToken(SourceMgr, retType->getEndLoc());
     SourceLoc throwsEndLoc = Lexer::getLocForEndOfToken(SourceMgr, throwsLoc);
-    (*diagOpt).fixItInsert(arrowLoc, rethrows ? "rethrows " : "throws ")
-              .fixItRemoveChars(typeEndLoc, throwsEndLoc);
+
+    if (isThrow) {
+      (*diagOpt).flush();
```

Comment:
Any particular reason you need the extra flush?

Reply:
Yes, without it got "Already have an active diagnostic" error. My understanding was that because i'm doing a diagnose of an error and then another diagnose of the note they were conflicting so I had to flush he first one. I found it done this way in other parts of the code.